p2p/peerexchange/bookchecker_test.go

```diff
@@ -154,6 +155,93 @@ func TestDiscovery_GetRandomPeers(t *testing.T) {
 	})
 }
 
+func TestHost_ReceiveAddressesOnCheck(t *testing.T) {
+	t.Parallel()
+
+	hosts, instances := setupOverNodes(t, 2)
+	nodeA, nodeB := instances[0], instances[1]
+
+	best, err := bestHostAddress(nodeA.host)
+	require.NoError(t, err)
+
+	connectedNodeAddress := addressFromHost(t, hosts[1])
+	nodeA.book.AddAddress(connectedNodeAddress, best)
+	wrapDiscovery(instances)
+
+	// check that after bootstrap, nodeB has only addresss of nodeA and nodeA has only addresss of nodeB
+	require.Eventually(t, func() bool {
+		aAddresses := nodeA.book.GetAddresses()
+		bAddresses := nodeB.book.GetAddresses()
+		if len(aAddresses) != 1 || len(bAddresses) != 1 {
+			return false
+		}
+		return aAddresses[0].ID.String() == nodeB.host.ID().String() && bAddresses[0].ID.String() == nodeA.host.ID().String()
```

Comment:
it seems this should be checked outside of require.Eventually(). once we are sure they each have size-1 address book, we should immediately exit this require.Eventually()